.github/workflows/release-facebook.yml

```diff
@@ -0,0 +1,123 @@
+
+            fs.writeFileSync('${{ env.RSS_FILENAME }}', JSON.stringify(data.items));
+        
+      - name: Caching Hash
+        id: blog-posts-cache
+        uses: actions/cache@v4
+        with:
+          path: |
+            ${{ env.RSS_FILENAME }}
+          key: blog-post-${{ hashFiles(env.RSS_FILENAME) }}          
+      
+      - name: Get latest posts
+#        if: ${{ steps.blog-posts-cache.outputs.cache-hit != 'true' }}
+        id: latest-data
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const fs = require('fs');
+            const data = fs.readFileSync('${{ env.RSS_FILENAME }}');
+            const items = JSON.parse(data)
+
+            let date
+            let photoUrl
+            let caption
+            for (const item of items) {
+              if (!date || new Date(date).getTime() < new Date(item.pubDate).getTime()) {
+                date = item.pubDate;
+                photoUrl = item.enclosure.link;
+                caption = `${item.title}\n\n${item.link}\n\n${item.description}`;
+              }
+            }
+
+            return JSON.stringify({photoUrl, caption});
+          result-encoding: string
+
+      - name: Get Facebook page access token
+#        if: ${{ steps.blog-posts-cache.outputs.cache-hit != 'true' }}
+        id: token
+        run: |
+          json=$(curl -X GET "https://graph.facebook.com/${{ env.FACEBOOK_USER_ID }}/accounts?access_token=${{ env.FACEBOOK_USER_ACCESS_TOKEN }}")            
+          token=$(echo $json | jq -r '.data[].access_token')
+          echo "::add-mask::$token"
+          echo "page_access_token=${token}" >> $GITHUB_OUTPUT
+
+      - name: Upload photo to Facebook (If published is 'true', the photo will be published on the page's timeline.)
+#        if: ${{ steps.blog-posts-cache.outputs.cache-hit != 'true' }}
+        id: photo_upload
+        run: |
+            photoUrl=$(echo '${{ steps.latest-data.outputs.result }}' | jq -r '.photoUrl')
+            caption=$(echo '${{ steps.latest-data.outputs.result }}' | jq -r '.caption')
+            json=$(curl -X POST \
+              -d "url=${photoUrl}" \
+              -d "caption=${caption}" \
+              -d "published=true" \
+              -d "access_token=${{ steps.token.outputs.page_access_token }}" \
+              "https://graph.facebook.com/${{ env.PAGE_ID }}/photos")
+              photo_id=$(echo $json | jq -r '.id')
+              echo "photo_id=${photo_id}" >> $GITHUB_OUTPUT
+    
+#       - name: Post feed to Facebook using the uploaded photo
+# #        if: ${{ steps.blog-posts-cache.outputs.cache-hit != 'true' }}
+#         run: |
+#             curl -i -X POST \
+#               -d "message=" \
+#               -d "attached_media[0]={"media_fbid":"${{ steps.photo_upload.outputs.photo_id }}"}" \
+#               -d "access_token=${{ steps.token.outputs.page_access_token }}" \
+#               "https://graph.facebook.com/${{ env.PAGE_ID }}/feed"
```

Comment:
Are these lines needed?

Reply:
Actually, it is not needed. I just kept it in case it can be useful later.

Reply:
Okay. For future reference, tell me what situation can it be useful for?

Reply:
This is not a photo upload, but a post to a feed with multiple images (`attached_media[0]`, `attached_media[1]`, and so on) and having one single common message.
I tried to use it when I couldn't manage to publish only photos. It worked but later I didn't need it any more because the `/photo` API was enough.

Reply:
I see. Could you add some comments to describe the use case?

Reply:
I added some comments to describe it.